Alignment/affinealignment.h

```diff
@@ -5,10 +5,14 @@
 #include "alignment.h"
 #include <limits>
 
+// TODO: needs more comments!
+// TODO: the similarity matrix should also be an object that knows about its size (n_rows, n_columns)
+// TODO: why not return an affine alignment object? instead of making the fxn void
 void doAffineAlignment(float *s, int signalA_len, int signalB_len, float go, float ge, bool OverlapAlignment, AffineAlignObj &affineAlignObj);
```

Comment:
explain what the parameters are, see for example https://github.com/OpenMS/OpenMS/blob/develop/src/openms/include/OpenMS/KERNEL/MSSpectrum.h#L296

```suggestion
    /**
      @brief Perform affine alignment
      
       Performs affine alignment using a given similarity matrix and returns score matrices
       
      @param[in] s Similarity matrix
      @param signalA_len ...
      @param signalB_len ...
      @param go Gap opening  ...
      @param ge Gap opening  ...
      @param[out] affineAlignObject Resulting affine alignment object containing the filled matrices
      
      @note Make sure that the similarity matrix size matches the size of the AffineAlignObj
      */
void doAffineAlignment(float *s, int signalA_len, int signalB_len, float go, float ge, bool OverlapAlignment, AffineAlignObj &affineAlignObj);
```

Reply:
Also, if the similiary matrix will not change, make it `const` to make that clear